### PR TITLE
Cf payment plan

### PR DESCRIPTION
**Pull request by connorfrendt**
Need someone to review the work I've done and see if it's what I should be doing for the task that is assigned to me

### CODE REVIEW

File: hifireg/registration/urls.py [hunk 1]
```diff
@@ -33,6 +33,7 @@
     path('register/miscellaneous/', views.register_misc, name='register_misc'),
 
     # payment:
+    path('payment/hello-world', views.registration.hello_world, name='hello_world'),
```
Comment (bryant-ferguson) [inline hifireg/registration/urls.py:36]: You don't need the "registration" infix here because we automatically import the contents of the registration folder into the "views" module. Also, all of these function based views (FBVs) have begun conversion to class based views (CBVs).
https://docs.djangoproject.com/en/3.0/topics/class-based-views/intro/

You may be better off just creating a new branch on origin/develop and copying your changes there and making sure to create this as a CBV.

Also, I think that making a "hello world" page in this case was a euphemism for creating a very basic page. So, that basic page should still have a name the is relevant to the task at hand. i.e. "PaymentPlan" 

So, to sum up, you'll add something like this to urls.py:
```
    path('payment/plan', views.PaymentPlan.as_view(), name='payment_plan'),
```

I think we could use a discussion on whether there is value in changing the root URL here to "payment" since ultimately this is still just part of "registration"... but that doesn't much matter for this task.

File: hifireg/registration/views/registration.py [hunk 2]
```diff
@@ -280,11 +280,15 @@ def make_payment(request):
     form = None
     if request.method == 'POST':
         if 'previous' in request.POST:
-            return redirect('register_misc')
+            # return redirect('register_misc')
+            return redirect('hello_world')
```
Comment (bryant-ferguson) [inline hifireg/registration/views/registration.py:284]: Looks good, but since I suggested renaming this url above, you'll use `payment_plan` instead.

Also, I understand the desire to preserve the previous state with a comment, but it actually results in a messier diff, so I recommend just deleting the commented out line. It's stored in git if we ever want to retrieve it.

File: hifireg/registration/views/registration.py [hunk 3]
```diff
@@ -280,11 +280,15 @@ def make_payment(request):
...
         return redirect('payment_confirmation')
     else:
         return render(request, 'registration/payment.html', {'form': form})
 
 
 def payment_confirmation(request):
     return render(request, 'registration/payment_confirmation.html', {})
+
+def hello_world(request):
+    return render(request, 'registration/base.html', {})
```
Comment (bryant-ferguson) [inline hifireg/registration/views/registration.py:294]: You want to use a CBV here instead.
```suggestion
class PaymentPlan(View):
    def get(self, request):
        return render(request, 'registration/base.html', {})

```
but that's still using "base.html" directly. 
What you want to do is inherit from "base.html". So you'll need to make a new template file.
Here is a resource for "django template inheritance":
https://docs.djangoproject.com/en/3.0/ref/templates/language/#template-inheritance

File: hifireg/registration/views/registration.py (hunk 2 above)
Comment (bryant-ferguson) [reply hifireg/registration/views/registration.py:284]: Oh, also at line 272, you need to replace 'payment' with the url to your new page since that's the redirect that will lead to your page.

**Review by bryant-ferguson — COMMENTED** [review]
Left a bunch of notes. 

I think you're better of creating a new branch at origin/develop since there have been significant changes to these files over there. Just copy the changes into that new branch.

**Comment by bryant-ferguson** [comment]
Closing, since this was done in another branch.
